src/Image.php

```diff
@@ -0,0 +1,269 @@
+		$this->resizeImage($options['height'], $options['width'], $options['crop']);
+
+		return $this;
+	}
+
+	/**
+	 * Convert to blob
+	 *
+	 * @return string
+	 */
+	public function toBlob()
+	{
+		return $this->image_handler->getImageBlob();
+	}
+
+	/**
+	 * Clear the image resource
+	 *
+	 * @return $this
+	 */
+	public function clear()
+	{
+		// Free our image resizer of our image resource, to enhance performance and make it easier to resize the NEXT image
+		// ( Garbage collect )
+		$this->image_handler->clear();
+
+		return $this;
+	}
+
+	/**
+	 * Resize the image
+	 *
+	 * @param int  $height
+	 * @param int  $width
+	 * @param bool $crop
+	 * @return $this|void
+	 */
+	public function resizeImage($height, $width, $crop = false)
+	{
+		if ($this->crop || $crop) {
+			$this->image_handler->cropThumbnailImage($width, $height);
+
+			return $this;
+		}
+
+		$this->image_handler->thumbnailImage($width, $height, true); // Best fit
+
+		return $this;
+	}
+
+	/**
+	 * Read an image from a blob
+	 *
+	 * @param string $blob
+	 * @return $this
+	 */
+	public function readFromBlob($blob)
+	{
+		$this->image_handler->readImageBlob($blob);
+
+		return $this;
+	}
+
+	/**
+	 * Set the image format
+	 *
+	 * @param string $format
+	 * @return $this
+	 */
+	public function setImageFormat($format)
+	{
+		$this->image_handler->setImageFormat($format);
+
+		return $this;
+	}
+
+	/**
+	 * Set the handler instance
+	 *
+	 * @param \Imagick $handler
+	 * @return $this
+	 */
+	public function setImageHandler(Imagick $handler)
+	{
+		$this->image_handler = $handler;
+
+		return $this;
+	}
+
+	/**
+	 * Set the compression quality
+	 *
+	 * @param int  $min
+	 * @param int  $max
+	 * @param bool $maximum_file_size
+	 * @return $this
+	 */
+	public function setQualityLevel($min = self::FULL_COMPRESSION, $max = self::NO_COMPRESSION, $maximum_file_size = false)
+	{
+		if (! $maximum_file_size) {
+			$this->image_handler->setImageCompressionQuality($max);
+
+			return $this;
+		}
+
+		$quality   = $max;
+		$starting_size = strlen($this->toBlob());
+		$size          = $starting_size;
+
+		do {
+			$this->image_handler->setImageCompressionQuality($quality);
+			$size = strlen($this->toBlob());
+			$quality--;
+		} while ($quality <= $max && $quality >= $min && $size > $maximum_file_size);
+
+		// If we've gotten this far and have increased the file size, just use the minimum compression (max quality)
+		// because compression is not useful in this situation.
+		if ($size >= $starting_size) {
+			$this->image_handler->setImageCompressionQuality($max);
+		}
+
+		return $this;
+	}
+
+	/**
+	 * Retrieve the Imagick instance
+	 *
+	 * @return \Imagick
+	 */
+	public function getImageHandler()
+	{
+		return $this->image_handler;
+	}
+
+	/**
+	 * Ensure that the request contains valid options
+	 *
+	 * @param array $options
+	 * @return array
+	 */
+	private function validateOptions(array $options)
```

Comment:
I do found these functions a bit oddly placed. Perhaps you could explain the reasoning to me at some point.